src/components/FeatureViews/AdminPortal/PanelManager.js

```diff
@@ -281,8 +291,18 @@ const editRecordRow = (record) => {
 }
 
 
-
-
+const getRecordList = () => {
+    const rList = [];
+    for(let i = 0; i <= directoryData["ids"].length; i++ ) {
+        const record = Object.values(directoryData).map(obj => {
+            return obj.values[i];
+        });
```

Comment:
This looks very familiar to your "getRecord()" function. Is it possible to delete the other function and just use this function? You can find the record list you're looking for then index into it? 

Still not very certain what you're trying to do for lines 297 - 299... you're saving record as a list of values from each directory data. Consider using different names or providing additional documentation.